### PR TITLE
Add a saving/loading demo with various serialization formats

### PR DESCRIPTION
**Pull request by Calinou**
This demo showcases how to save a simple game with ConfigFile and JSON.

### CODE REVIEW

File: misc/save_load/player.gd [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+func _process(delta):
+	# Player movement (controller-friendly).
+	motion.x += Input.get_action_strength("move_right") * MOVE_SPEED * delta
+	motion.x -= Input.get_action_strength("move_left") * MOVE_SPEED * delta
+	motion.y += Input.get_action_strength("move_down") * MOVE_SPEED * delta
+	motion.y -= Input.get_action_strength("move_up") * MOVE_SPEED * delta
+	# Apply friction.
+	motion *= 0.9
+
+	position += motion
```
Comment (aaronfranke) [inline misc/save_load/player.gd:20]: The motion overcomplicates the demo, and also you currently aren't saving the motion anyway, just the position.

```suggestion
	var velocity = Vector2.ZERO
	velocity.x = Input.get_action_strength("move_right") - Input.get_action_strength("move_left")
	velocity.y = Input.get_action_strength("move_down") - Input.get_action_strength("move_up")
	position += velocity * MOVE_SPEED * delta
```

**Review by aaronfranke — CHANGES_REQUESTED** [review]

**Comment by Calinou** [comment]
I applied the suggestions, removed reference to documentation not available yet and added a screenshot.

File: misc/save_load/README.md [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Saving and Loading (Serialization)
+
+This demo showcases how to save a simple game with each serialization
+format supported by Godot:
+
+- ConfigFile
+- JSON
+
+More formats may be added in the future.
+
```
Comment (aaronfranke) [inline misc/save_load/README.md:10]:
```suggestion

For more information, see this documentation article:
https://docs.godotengine.org/en/latest/tutorials/io/saving_games.html

```

File: misc/save_load/enemy.gd [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+	# warning-ignore:return_value_discarded
+	move_and_slide(Vector2(MOVE_SPEED, 0))
+
+	# The enemy went outside of the window. Move it back to the left.
+	if position.x >= 732:
+		position.x = -32
+
+
+func _on_AttackArea_body_entered(body: Node) -> void:
```
Comment (aaronfranke) [inline misc/save_load/enemy.gd:23]: For now the demos don't use static types per Akien's request.

```suggestion
func _on_AttackArea_body_entered(body):
```

File: misc/save_load/enemy.gd [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+	if body.name == "Player":
+		attacking = body
+
+
+func _on_AttackArea_body_exited(_body: Node) -> void:
```
Comment (aaronfranke) [inline misc/save_load/enemy.gd:28]:
```suggestion
func _on_AttackArea_body_exited(_body):
```

File: misc/save_load/save_load_config_file.gd [hunk 5]
```diff
@@ -0,0 +1,50 @@
+extends Button
+
```
Comment (aaronfranke) [inline misc/save_load/save_load_config_file.gd:2]:
```suggestion
# This script shows how to save data using Godot's custom ConfigFile format.
# ConfigFile can store any Godot type natively.

```

File: misc/save_load/save_load_json.gd [hunk 6]
```diff
@@ -0,0 +1,58 @@
+extends Button
+
```
Comment (aaronfranke) [inline misc/save_load/save_load_json.gd:2]:
```suggestion
# This script shows how to save data using the JSON file format.
# JSON is a widely used file format, but not all Godot types can be
# stored natively. For example, integers get converted into doubles,
# and to store Vector2 and other Variant types you need `var2str()`.

```

**Review by aaronfranke — CHANGES_REQUESTED** [review]
I think this should be moved to the `2d/` folder. While this isn't a demo "for 2D", it is a demo that uses Node2D, and I moved similar demos such as Instancing and Tween to that folder.

**Review by aaronfranke — CHANGES_REQUESTED** [review]
Looks good, but there is one last (fairly important) thing: The `screenshots/` folder is missing a `.gdignore` file. We should add one to prevent Godot from importing `screenshots/save_load.png`.

**Comment by Calinou** [comment]
@aaronfranke The option appears to be enabled here:

image

**Review by aaronfranke — APPROVED** [review]
> The option appears to be enabled here:

Weird, maybe you have something on your account that prevents it? All I know is that I can't push to your repo.

**Pull request merged**